src/engine/guidance/assemble_overview.cpp

```diff
@@ -85,16 +59,37 @@ std::vector<util::Coordinate> assembleOverview(const std::vector<LegGeometry> &l
     std::vector<util::Coordinate> overview_geometry;
     overview_geometry.reserve(overview_size);
 
+    using GeometryIter = decltype(overview_geometry)::const_iterator;
+
     auto leg_reverse_index = leg_geometries.size();
-    for (const auto &geometry : leg_geometries)
-    {
-        auto begin = geometry.locations.begin();
-        auto end = geometry.locations.end();
+    const auto insert_without_overlap = [&leg_reverse_index, &overview_geometry](GeometryIter begin, GeometryIter end) {
+        // not the last leg
         if (--leg_reverse_index > 0)
         {
             end = std::prev(end);
         }
         overview_geometry.insert(overview_geometry.end(), begin, end);
+    };
+
+    if (use_simplification)
+    {
+        const auto zoom_level = std::min(18u, calculateOverviewZoomLevel(leg_geometries));
+        for (const auto &geometry : leg_geometries)
+        {
+            auto simplified = douglasPeucker(geometry.locations.begin(), geometry.locations.end(), zoom_level);
+            auto begin = simplified.cbegin();
+            auto end = simplified.cend();
+            insert_without_overlap(begin, end);
+        }
+    }
+    else
+    {
+        for (const auto &geometry : leg_geometries)
+        {
+            auto begin = geometry.locations.begin();
+            auto end = geometry.locations.end();
+            insert_without_overlap(begin, end);
```

Comment:
Problem with that is that `douglasPeucker` returns a rvalue and we would need to add a copy for `leg_geometries` to get the same interface (we can't move, since we are not allowed to modify `leg_geometries`).

I would prefer to keep the current interface since it is pretty obvious what it does and works well.